vm/src/obj/objtype.rs

```diff
@@ -494,9 +516,31 @@ pub fn new(
     _base: PyClassRef,
     bases: Vec<PyClassRef>,
     dict: HashMap<String, PyObjectRef>,
+    optional_vm: Option<&VirtualMachine>,
```

Comment:
Not sure about this... What do you think about doing the unique bases check in the methods that call this instead of here? Maybe this function will return a `Result` instead of a `PyResult`?

Reply:
Returning a result here and unwrapping on caller if he doesn't have a vm is much cleaner. Good idea.